parser/class_files.py

```diff
@@ -31,6 +30,39 @@ def process_c_file(class_file_src_path, class_file_target_path, class_name, file
     # Parse the images
     parse_images(html_soup, dita_body, dita_soup)
 
+    # Parse the summary block
+    if  html_soup.find("td", {"colspan": "6"}) is None:
+        print(f">>> Failed to find colspan:6 for {class_file_src_path}")
+    else:
+         options = {'file_name': file_name, 'file_path': class_file_src_path}
+
+         # Parse the summary and the signature blocks from the html and build a dita <signagures> and <summary> blocks
+         parse_summary_and_signatures(html_soup, dita_body, dita_soup)
+
+         # Parse the propulsion block and build a dita <propulsion>
```

Comment:
@AbegaM - propulsion and remarks sections are not dependent on there being a colspan:6 table. 

If there isn't a colspan:6 table, we can't do `summary and signatures`, but we can still handle propulsion and remarks.